scripts/unix_share_build.sh

```diff
@@ -131,66 +154,66 @@ function build_dagmc(){
+        $SUDO cp -pv CubitUtilConfig.cmake CubitUtilConfig.cmake.orig
+        $SUDO $SED -i "s/\/\.\.\/app_logger\;//" CubitUtilConfig.cmake
     fi
 
-    cd ${CUBIT_PATH}
-    if [ "${1}" = "2020.2" ]; then
-        CUBIT_BASE_NAME="Coreform-Cubit-2020.2"
-    elif [ "${1}" = "17.1.0" ]; then
-        CUBIT_BASE_NAME="Trelis-17.1"
+    cd /Applications
+
+    # 17.1.0 comes with a separate SDK. It is unclear yet on how it is supposed to be installed and used.
+    # this is a way to have it working... 
+    if [ "$1" == "17.1.0" ] ; then
+        cd ${CUBIT_PATH}
+        $SUDO tar -xzf ${FOLDER_PKG}/${CUBIT_SDK_PKG}
+        $SUDO rsync -a  ${CUBIT_BASE_NAME}/* ./
+        $SUDO rsync -a  ${CUBIT_BASE_NAME}.app/Contents/MacOS/* MacOS/
+        $SUDO rsync -a bin/* MacOS/
+        $SUDO rm -rf bin ${CUBIT_BASE_NAME}.app
+        $SUDO ln -s MacOS bin
+        $SUDO ln -s ${CUBIT_PATH}/include /Applications/include
+
+        #  # fixing the path to Contents/Include
+        $SUDO cp -pv ${CUBIT_PATH}/MacOS/CubitExport-release.cmake ${CUBIT_PATH}/MacOS/CubitExport-release.cmake.orig
+        $SUDO $SED -i "s/\${_IMPORT_PREFIX}/\/Applications/" ${CUBIT_PATH}/MacOS/CubitExport-release.cmake
     fi
-    sudo tar -xzf ${FOLDER_PKG}/${CUBIT_SDK_PKG}
-    sudo mv ${CUBIT_BASE_NAME}/* ./
-    sudo mv ${CUBIT_BASE_NAME}.app/Contents/MacOS/* MacOS/
-    sudo mv bin/* MacOS/
-    sudo rm -rf bin ${CUBIT_BASE_NAME}.app
-    sudo ln -s MacOS bin
-    sudo ln -s ${CUBIT_PATH}/include /Applications/include
-
 
-    # fixing the path to Contents/Include
-    sudo cp -pv ${CUBIT_PATH}/MacOS/CubitExport-release.cmake ${CUBIT_PATH}/MacOS/CubitExport-release.cmake.orig
-    sudo gsed -i "s/\/${CUBIT_BASE_NAME}.app\/Contents//" ${CUBIT_PATH}/MacOS/CubitExport-release.cmake
+    hdiutil detach /Volumes/Cubit
 
-    sudo cp -pv ${CUBIT_PATH}/MacOS/CCubitGeomConfig.cmake ${CUBIT_PATH}/MacOS/CubitGeomConfig.cmake.orig
-    sudo gsed -i "s/\${_IMPORT_PREFIX}\/include/\${_IMPORT_PREFIX}\/${CUBIT_BASE_NAME}.app\/include/" ${CUBIT_PATH}/MacOS/CubitGeomConfig.cmake
-
-
-    # removing app_loger that seems to not be present in Cubit 2020.2
-    if [ "${1}" = "2020.2" ]; then
-        cd ${CUBIT_PATH}/bin
-        sudo cp -pv CubitExport.cmake CubitExport.cmake.orig
-        sudo gsed -i "s/\"\/\.\.\/app_logger\"/\"\"/" CubitExport.cmake
-        sudo cp -pv CubitUtilConfig.cmake CubitUtilConfig.cmake.orig
-        sudo gsed -i "s/\/\.\.\/app_logger\;//" CubitUtilConfig.cmake
-    fi
 
 }
 
-function linux_setup_cubit_sdk() {
+function linux_setup_cubit() {
 
+  
     cd ${FOLDER_PKG}
     $SUDO apt-get install -y ./${CUBIT_PKG}
+
+    if [ "$1" == "2021.3" ] || [ "$1" == "2021.4" ] ; then
+	    return
+    fi
+
     cd /opt
     $SUDO tar -xzf ${FOLDER_PKG}/${CUBIT_SDK_PKG}
     # removing app_loger that seems to not be present in Cubit 2020.2
     if [ "$1" = "2020.2" ]; then
         cd ${CUBIT_PATH}/bin
         $SUDO cp -pv CubitExport.cmake CubitExport.cmake.orig
```

Comment:
The lines inside the if statement are identical to the ones in the `mac_setup_cubit` function. IMO it would be good to separate these commands into their own function so we don't miss an update to the regex in one or the other.

Reply:
done

thx for the review !